src/lua/datetime.lua

```diff
@@ -680,11 +1114,62 @@ ffi.metatype(datetime_t, {
     __eq = datetime_eq,
     __lt = datetime_lt,
     __le = datetime_le,
+    __sub = datetime_interval_sub,
+    __add = datetime_interval_add,
     __index = datetime_index,
 })
 
-return setmetatable({
-    new         = datetime_new,
-    now         = datetime_now,
-    is_datetime = is_datetime,
-}, {})
+local function total_secs(self)
+    return self.sec + self.nsec / 1e9
+end
+
+local interval_index_fields = {
+    usec = function(self) return math_floor(self.nsec / 1e3) end,
+    msec = function(self) return math_floor(self.nsec / 1e6) end,
+
+    week = function(self)
+        return (math_modf(total_secs(self) / (7 * SECS_PER_DAY)))
+    end,
+    day = function(self)
+        return (math_modf(total_secs(self) / SECS_PER_DAY))
+    end,
+    hour = function(self)
+        return (math_modf(total_secs(self) / (60 * 60)))
+    end,
+    min =  function(self)
+        return (math_modf(total_secs(self) / 60))
```

Comment:
You don't need the `()` around returned values. In all places.

Reply:
Done